modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala

```diff
@@ -0,0 +1,94 @@
+import lucuma.react.common.*
+import lucuma.react.primereact.Button
+import lucuma.react.primereact.InputText
+import lucuma.react.primereact.OverlayPanel
+import lucuma.react.primereact.SelectButtonOptional
+import lucuma.react.primereact.SelectItem
+import lucuma.react.primereact.hooks.all.*
+import lucuma.schemas.model.Dataset
+import lucuma.ui.primereact.*
+import lucuma.ui.sequence.given
+import lucuma.ui.syntax.render.*
+import observe.ui.Icons
+import observe.ui.ObserveStyles
+import observe.ui.model.EditableQaFields
+import org.scalajs.dom.KeyCode
+
+case class QaEditor(
+  dataset:    Dataset,
+  renderIcon: VdomNode,
+  onChange:   Dataset.Id => EditableQaFields => Callback
+) extends ReactFnProps(QaEditor.component):
+  val editableQaFields: EditableQaFields = EditableQaFields.fromDataset.get(dataset)
+
+object QaEditor:
+  private type Props = QaEditor
+
+  private val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      .useStateViewBy(_.editableQaFields)
+      .useOverlayPanelRef
+      .render: (props, qaFields, panelRef) =>
+        val submit: Callback = props.onChange(props.dataset.id)(qaFields.get)
+        val reset: Callback  = qaFields.set(props.editableQaFields)
+
+        <.span(ObserveStyles.QaStatusEditable, ^.onClick ==> panelRef.toggle)(
+          props.renderIcon,
+          <.span(ObserveStyles.QaStatusSelect)(Icons.ChevronDown),
+          OverlayPanel(onHide = reset)(
+            <.form(ObserveStyles.QaEditorPanel)(
+              ^.onSubmit --> submit,
+              ^.onKeyDown ==> { e =>
+                e.keyCode match // Allows submitting by pressing enter after changing the status.
+                  case KeyCode.Enter => submit >> panelRef.hide
+                  case _             => Callback.empty
+              }
+            )(
+              SelectButtonOptional[DatasetQaState](
+                clazz = ObserveStyles.QaStatusButtonStrip,
+                value = qaFields.get.qaState,
+                options = Enumerated[DatasetQaState].all.map: qaState =>
+                  SelectItem(qaState, qaState.tag),
+                onChange = value => qaFields.mod(EditableQaFields.qaState.replace(value)),
+                itemTemplate = qaState =>
+                  React.Fragment(
+                    qaState.value.some.renderVdom,
+                    qaState.value.tag
+                  )
+              ).withMods(
+                ^.onClick ==> ((e: ReactEvent) => e.stopPropagationCB)
+              ).compact,
+              InputText(
+                id = "qaComment",
+                value = qaFields.get.comment.map(_.value).orEmpty,
+                placeholder = "Comment",
+                onChange = e =>
+                  qaFields.mod:
+                    EditableQaFields.comment.replace(NonEmptyString.from(e.target.value).toOption)
+              ).withMods(
+                ^.autoComplete.off,
+                ^.onClick ==> ((e: ReactEvent) => e.stopPropagationCB)
```

Comment:
does `_.stopProp..` work?

Reply:
Actually, it does work but you have to wrap in parens; ie: `^.onClick ==> (_.stopPropagationCB)`. I've pushed a commit.